app/services/work_uploads_edit_service.rb

```diff
@@ -21,32 +27,25 @@ def find_post_curation_uploads(work:, upload_keys: [])
 
       private
 
-        def replace_uploads(persisted_pre_curation_uploads, replaced_uploads_params)
-          updated_uploads = []
-          persisted_pre_curation_uploads.each_with_index do |existing, i|
+        def replace_uploads(work, replaced_uploads_params)
+          new_uploads = []
+          work.pre_curation_uploads.each_with_index do |existing, i|
             key = i.to_s
-
-            if replaced_uploads_params.key?(key)
-              replaced = replaced_uploads_params[key]
-              updated_uploads << replaced
-            else
-              updated_uploads << existing.blob
-            end
+            next unless replaced_uploads_params.key?(key)
+            new_uploads << replaced_uploads_params[key]
+            existing.blob.purge
+            existing.purge
```

Comment:
This is not a criticism, but I believe that this might be redundant (please see https://github.com/rails/rails/blob/main/activestorage/app/models/active_storage/attachment.rb#L32). I suspect that `existing.purge` should also invoke `purge` for the associated `blob`.

Reply:
@jrgriffiniii yes, I think you are correct.  The reload was needed and this was put in to try and figure out why the Purge Job was still getting called before I put in the reload.   I will take it out.